.github/workflows/main.yml

```diff
@@ -0,0 +1,331 @@
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+# or visit www.oracle.com if you need additional information or have any
+# questions.
+#
+
+name: 'OpenJDK GHA Sanity Checks'
+
+on:
+  push:
+    branches-ignore:
+      - master
+      - pr/*
+  workflow_dispatch:
+    inputs:
+      platforms:
+        description: 'Platform(s) to execute on (comma separated, e.g. "linux-x64, macos, aarch64")'
+        required: true
+        default: 'linux-x64, linux-x86, linux-x64-variants, linux-cross-compile, macos-x64, macos-aarch64, windows-x64, windows-aarch64'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: false
+
+jobs:
+
+  ###
+  ### Determine platforms to include
+  ###
+
+  select:
+    name: 'Select platforms'
+    runs-on: ubuntu-20.04
+    outputs:
+      linux-x64: ${{ steps.include.outputs.linux-x64 }}
+      linux-x86: ${{ steps.include.outputs.linux-x86 }}
+      linux-x64-variants: ${{ steps.include.outputs.linux-x64-variants }}
+      linux-cross-compile: ${{ steps.include.outputs.linux-cross-compile }}
+      macos-x64: ${{ steps.include.outputs.macos-x64 }}
+      macos-aarch64: ${{ steps.include.outputs.macos-aarch64 }}
+      windows-x64: ${{ steps.include.outputs.windows-x64 }}
+      windows-aarch64: ${{ steps.include.outputs.windows-aarch64 }}
+
+    steps:
+        # This function must be inlined in main.yml, or we'd be forced to checkout the repo
+      - name: 'Check what jobs to run'
+        id: include
+        run: |
+          # Determine which platform jobs to run
+
+          # Returns 'true' if the input platform list matches any of the platform monikers given as argument,
+          # 'false' otherwise.
+          # arg $1: platform name or names to look for
+          function check_platform() {
+            if [[ '${{ !secrets.JDK_SUBMIT_FILTER || startsWith(github.ref, 'refs/heads/submit/') }}' == 'false' ]]; then
+              # If JDK_SUBMIT_FILTER is set, and this is not a "submit/" branch, don't run anything
+              echo 'false'
+              return
+            fi
+
+            if [[ $GITHUB_EVENT_NAME == workflow_dispatch ]]; then
+              input='${{ github.event.inputs.platforms }}'
+            elif [[ $GITHUB_EVENT_NAME == push ]]; then
+              input='${{ secrets.JDK_SUBMIT_PLATFORMS }}'
+            else
+              echo 'Internal error in GHA'
```

Comment:
@magicus isn't this check too strict? The Corretto team runs these steps for all our PRs to make sure we will not break the OpenJDK later if our fix will be proposed upstream. But this line prevents us to run it by the "pull_request". I can apply the local patch for sure, but probably this check can be relaxed?
Something like:
```
            elif [[ $GITHUB_EVENT_NAME == pull_request ]]; then
              input='${{ secrets.JDK_SUBMIT_PLATFORMS }}'
            else
```

Reply:
@mrserb possibly related: [JDK-8291444](https://bugs.openjdk.org/browse/JDK-8291444)